### DISCUSSION TITLE
Guidance on Proper Attribution for Customized Versions of RagApp under Apache License 2.0

### DISCUSSION THREAD

**Discussion by Rauljauregi** [ragapp/ragapp#178] [Q&A]
First of all, thank you for the excellent work on RagApp. I'm currently integrating it into my stack, and as part of that process, I want to ensure that I am fully compliant with the requirements of the Apache License 2.0, under which RagApp is distributed.

Areas for Clarification:

**Attribution Requirements:** I have customized RagApp to better fit the needs of my end users, including changes to the user interface and functionality. Could you please provide detailed guidance on how to correctly attribute the original RagApp project according to the Apache License 2.0? Specifically, I would like to know:

The best practices for including attribution notices in the codebase, especially for modified files.
Recommendations for including an attribution in the user interface, such as in the "About" section or other relevant areas.
Any additional steps I should take to ensure full compliance when distributing the customized version to my clients.
Documentation of Modifications: As I've made modifications to the source code, I understand that I need to include prominent notices indicating these changes. Could you advise on the most effective way to document and communicate these modifications within the code and documentation?

**Expected Outcome:** My goal is to ensure that I am fully compliant with the attribution requirements of the Apache License 2.0 while distributing a customized version of RagApp. Clear guidance on how to properly credit the original authors and document my changes would be greatly appreciated.

**Comment by marcusschiesser** [comment]
Here's some good information of what to do: https://opensource.stackexchange.com/questions/5143/how-to-mark-a-copied-apache-v2-piece-of-code
